[Concept Entry] Created Terms folder and added Euclidean algorithm

dakshdeepHERE opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->
Fixes #2454

dakshdeepHERE commented:
@SSwiniarski No changes required from your side ?

SSwiniarski commented:
@dakshdeepHERE Have patience, it hasn't been reviewed yet.

CBID2 commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,152 @@
+
+## Method 2
+
+In this method, we aim to compare both 'a' and 'b'. Whichever number is greater, we subtract the smaller number from the larger number and update the value of the larger number accordingly. If 'a' is greater than 'b', we replace 'a' with 'a - b', and if 'b' is greater than 'a', we replace 'b' with 'b - a'. We repeat this step until 'a' becomes equal to 'b'. Once 'a' and 'b' are equal, we return ' ...
```
[Nitpick] According to the docs style guide, pronouns can't be used. 
```suggestion
In this method, the aim is to compare both 'a' and 'b'. Whichever number is greater,  the smaller number is subtracted from the larger number. It also updates the value of the larger number accordingly. If 'a' is greater than 'b', replace 'a' with 'a - b', and if 'b' is greater than 'a', replace 'b' with 'b - a'. Repeat this step until 'a' becomes equal to 'b'. Once 'a' and 'b' are equal, return 'a' as the answer. At this point, the value of 'a' (which is equal to 'b') represents the highest common factor (HCF) of the original values of 'a' and 'b'.
```

CBID2 commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,152 @@
+  - 'paths/computer-science'
+---
+
+The `Euclidean algorithm` is a recursive algorithm that allows us to find the highest common factor (HCF) of two numbers. It is based on the observation that if we have two numbers, say A and B, and if B divides A evenly (leaving no remainder), then B is the HCF of A and B. In other words, the larger number is divisible by the smaller number without any remainder.
```
[Nitpick] See my feedback on [this line](https://github.com/Codecademy/docs/pull/2455#discussion_r1226023651). They both share the same problem. 
```suggestion
The `Euclidean algorithm` is a recursive algorithm that allows us to find the highest common factor (HCF) of two numbers. It is based on the observation that if we have two numbers, say A and B, and if B divides A evenly (leaving no remainder), then B is the HCF of A and B. In other words, the larger number is divisible by the smaller number without any remainder.
```

CBID2 commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
````diff
@@ -0,0 +1,152 @@
+5
+```
+
+If we debug the code above step by steps then:
+
+- Since 'b' is greater than 'a' (20 > 15), we replace 'b' with 'b - a', which gives us 'b = 20 - 15 = 5'.
+- Now 'a' is still 15, and 'b' is 5.
+- We proceed to replace 'a' with 'a - b', resulting in 'a = 15 - 5 = 10'.
+- The values are now 'a = 10' and 'b = 5'.
+- We continue the process and replace 'a' with 'a - b', giving us 'a = 10 - 5 = 5'.
+- Now 'a' and 'b' are both equal to 5.
+- At this point, the while loop exits, and we return 'a' as our answer, which is 5.
+- Therefore, the highest common factor (HCF) of (15, 20) is 5.
+
+## Method 3
+
+In this method we will use [recursion](https://www.codecademy.com/learn/discrete-math-recurrence-relations). The method takes two integer parameters 'a' and 'b' . Then returns an integer as the result. The gcd method uses a `recursive` approach to implement the `Euclidean algorithm`. It first checks if `b` is equal to 0. If it is, then it means that `a` is the GCD, and it returns `a` as the resul ...
````
[Nitpick] See my feedback on [this line](https://github.com/Codecademy/docs/pull/2455#discussion_r1226023651). They share the same problem. 
```suggestion
If we debug the code above step by steps then:

- Since 'b' is greater than 'a' (20 > 15), we replace 'b' with 'b - a', which gives us 'b = 20 - 15 = 5'.
- Now 'a' is still 15, and 'b' is 5.
- We proceed to replace 'a' with 'a - b', resulting in 'a = 15 - 5 = 10'.
- The values are now 'a = 10' and 'b = 5'.
- We continue the process and replace 'a' with 'a - b', giving us 'a = 10 - 5 = 5'.
- Now 'a' and 'b' are both equal to 5.
- At this point, the while loop exits, and we return 'a' as our answer, which is 5.
- Therefore, the highest common factor (HCF) of (15, 20) is 5.

## Method 3

In this method we will use [recursion](https://www.codecademy.com/learn/discrete-math-recurrence-relations). The method takes two integer parameters 'a' and 'b' . Then returns an integer as the result. The gcd method uses a `recursive` approach to implement the `Euclidean algorithm`. It first checks if `b` is equal to 0. If it is, then it means that `a` is the GCD, and it returns `a` as the result. If `b` is not 0, it means there is a remainder when `a` is divided by `b`. In this case, the method calls itself recursively with the arguments `b` and `a % b`. This recursive call continues until `b` becomes 0, at which point the base case is triggered, and the GCD is found.
```

CBID2 reviewed (changes requested):
Hi @dakshdeepHERE! :) I gave some feedback on your entry. I hope you find them helpful (note: I also recommend running your entry through Grammarly or spellcheck for errors! 😊🙏🏽

dakshdeepHERE commented:
Alright @SSwiniarski and @CBID2  I mostly try reverify everything through my grammarly extension already. Thanks for the advise is there anything else I can improve? ❤️🙏🏻

CBID2 commented:
> Alright @SSwiniarski and @CBID2 I mostly try reverify everything through my grammarly extension already. Thanks for the advise is there anything else I can improve? ❤️🙏🏻

Outside of updating your branch, nothing on my end @dakshdeepHERE. :)

dakshdeepHERE commented:
@CBID2 Check the changes I removed all the pronouns and refactored the sentences too with appropriate grammar. Let me know if further changes are required.

dakshdeepHERE commented:
Also, I need suggestions on the last part. In the future, I will continue following this trend with all my algorithm contributions. It is important to understand the differences between using different approaches and how they affect the respective time complexity.

CBID2 reviewed (changes requested):
The third part looks ok @dakshdeepHERE. I just gave a suggestion on how to improve the final paragraph.

dakshdeepHERE commented:
Done @CBID2

CBID2 commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+  - 'paths/computer-science'
+---
+
+The `Euclidean algorithm` is a recursive algorithm that allows us to find the highest common factor (HCF) of two numbers. It is based on the observation that if two numbers are given, say A and B, where B divides A evenly (leaving no remainder), then B is the HCF of A and B. In other words, the larger number is divisible by the smaller number without any remainder.
```
```suggestion
The `Euclidean algorithm` is a recursive algorithm that allows developers to find the highest common factor (HCF) of two numbers. It is based on the observation that if two numbers are given, say A and B, where B divides A evenly (leaving no remainder), then B is the HCF of A and B. In other words, the larger number is divisible by the smaller number without any remainder.
```

CBID2 commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+## Method 1
+
+In the basic approach, Let's follow a simple rule to find the GCD. Let's start by finding the minimum value between the two given numbers. Then, let's divide this minimum value by both numbers. If the number does not divide both numbers evenly, let's decrease it by `1` and continue dividing it by both numbers. Let's repeat this process until the minimum value can divide both numbers evenly. At th ...
```
```suggestion
In the basic approach, the purpose is to find the GCD. To do this, find the minimum value between the two given numbers. Then, divide it by both numbers. If the number does not divide both numbers evenly, decrease it by `1` and continue dividing. Lastly, repeat this process until the minimum value can divide both numbers evenly. At this point, the minimum value is the HCF (highest common factor).
```

CBID2 commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+In this method, a recursive approach is used to implement the Euclidean algorithm for finding the greatest common divisor (GCD) of two integers, 'a' and 'b'. The method takes 'a' and 'b' as integer parameters and returns an integer as the result. The GCD method starts by checking if 'b' is equal to 0. If it is, then it means that 'a' is the GCD, and it returns 'a' as the result. However, if 'b' i ...
+
+For more information on recursion, you can refer to this [resource](https://www.codecademy.com/learn/java-algorithms/modules/recursion-apcs/cheatsheet).
```
```suggestion
For more information on recursion, refer to this [resource](https://www.codecademy.com/learn/java-algorithms/modules/recursion-apcs/cheatsheet).
```

CBID2 reviewed (changes requested):
Just found some more things to improve @dakshdeepHERE. Keep it up! :)

CBID2 reviewed (approved):
Looks good to me

dakshdeepHERE commented:
Thanks for the suggestions @CBID2 My next submissions won't have these pronouns.

dakshdeepHERE commented:
Sorry for the tag, Can you look into this @SSwiniarski @caupolicandiaz its been like a week.

SSwiniarski commented:
@dakshdeepHERE I'm looking into this now.

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+---
+Title: 'Euclidean algorithm'
+Description: 'The Euclidean algorithm is a simple and efficient method for finding the highest common factor (HCF), also known as the greatest common divisor (GCD), of two numbers.'
```
```suggestion
Description: 'A simple and efficient method for finding the highest common factor (HCF), also known as the greatest common divisor (GCD), of two numbers.'
```
This because of the way terms are displayed on the parent page. They show up as [title] - [description] so having the title of the entry in the description will cause a redundancy.

dakshdeepHERE replied:
alright

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+  - 'Computer Science'
+Tags:
+  - 'Algorithms'
+  - 'Hcf'
```
```suggestion
  - 'Arithmetic'
```
Reminder, tags need to come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).

dakshdeepHERE replied:
Got it. We have to open a separate PR to add tags in tag.md?

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+- Method 1:
+
+  - The first code uses a simple iterative approach to find the greatest common divisor (GCD) of two numbers. It starts by finding the minimum value between `a` and `b`, then iterates from that value down to 1, checking if it divides both `a` and `b`. Therefore, the time complexity of this code is O(min(a, b)).
```
```suggestion
  - The first code uses a simple iterative approach to find the greatest common divisor (GCD) of two numbers. It starts by finding the minimum value between `a` and `b`, then iterates from that value down to 1, checking if it divides both `a` and `b`. Therefore, the time complexity of this code is **O(min(a, b))**.
```

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+- Method 2:
+
+  - The second code also calculates the GCD using an iterative approach known as the Euclidean algorithm. It repeatedly subtracts the smaller number from the larger number until the two numbers become equal (the GCD). The time complexity of this algorithm depends on the number of iterations required to reach the GCD. In the worst case, where one number is a multiple of the other, the time complex ...
```
```suggestion
  - The second code also calculates the GCD using an iterative approach known as the Euclidean algorithm. It repeatedly subtracts the smaller number from the larger number until the two numbers become equal (the GCD). The time complexity of this algorithm depends on the number of iterations required to reach the GCD. In the worst case, where one number is a multiple of the other, the time complexity is **O(max(a, b))**.
```

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+- Method 3:
+
+  - The third code is an optimized version of the Euclidean algorithm that uses recursion. It calculates the GCD by repeatedly taking the modulus of `a` with `b` and calling itself with the new values (`b` and `a%b`) until `b` becomes 0. The time complexity of this optimized Euclidean algorithm is also O(log(max(a, b))) since the algorithm reduces the values quickly by taking the modulus.
```
```suggestion
  - The third code is an optimized version of the Euclidean algorithm that uses recursion. It calculates the GCD by repeatedly taking the modulus of `a` with `b` and calling itself with the new values (`b` and `a%b`) until `b` becomes 0. The time complexity of this optimized Euclidean algorithm is **O(log(max(a, b)))** since the algorithm reduces the values quickly by taking the modulus.
```

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+To summarize, the time complexities of the three codes are as follows:
+
+- Code 1: O(min(a, b))
```
```suggestion
- Code 1: **O(min(a, b))**
```

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+To summarize, the time complexities of the three codes are as follows:
+
+- Code 1: O(min(a, b))
+- Code 2: O(max(a, b))
```
```suggestion
- Code 2: **O(max(a, b))**
```

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+- Code 1: O(min(a, b))
+- Code 2: O(max(a, b))
+- Code 3: O(log(max(a, b)))
```
```suggestion
- Code 3: **O(log(max(a, b)))**
```

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+- Code 2: O(max(a, b))
+- Code 3: O(log(max(a, b)))
+
+These time complexities typically represent the worst-case scenarios and assume that the `a` and `b` values are relatively large. In practice, the actual time taken by the algorithms can vary depending on the input values.
```
```suggestion
> **Note:** These time complexities typically represent the worst-case scenarios and assume that the `a` and `b` values are relatively large. In practice, the actual time taken by the algorithms can vary depending on the input values.
```

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
````diff
@@ -0,0 +1,149 @@
+
+In the basic approach, the purpose is to find the GCD. To do this, find the minimum value between the two given numbers. Then, divide it by both numbers. If the number does not divide both numbers evenly, decrease it by `1` and continue dividing. Lastly, repeat this process until the minimum value can divide both numbers evenly. At this point, the minimum value is the HCF (highest common factor).
+
+```java
````
Can we do these examples in pseudocode rather than Java?

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
````diff
@@ -0,0 +1,149 @@
+
+In this method, the aim is to compare both 'a' and 'b'. Whichever number is greater,  the smaller number is subtracted from the larger number. It also updates the value of the larger number accordingly. If 'a' is greater than 'b', replace 'a' with 'a - b', and if 'b' is greater than 'a', replace 'b' with 'b - a'. Repeat this step until 'a' becomes equal to 'b'. Once 'a' and 'b' are equal, return  ...
+
+```java
````
Can we do this in pseudocode rather than Java?

SSwiniarski commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
````diff
@@ -0,0 +1,149 @@
+
+For more information on recursion, refer to this [resource](https://www.codecademy.com/learn/java-algorithms/modules/recursion-apcs/cheatsheet).
+
+```java
````
Can we do this in pseudocode rather than Java?

SSwiniarski reviewed (changes requested):
@dakshdeepHERE I did a review and left a few comments. I also have a big overall comment: Because general entries are supposed to be language agnostic, can we write the examples in pseudocode rather than a specific programming language so this entry doesn't assume a knowledge of Java.

dakshdeepHERE commented:
@SSwiniarski Check the changes once to confirm if there's more changes needed. Also I will keep this check that I don't use tag **java** and instead of it use **pseudo** for examples. I thought the whole algorithm is coded in java so using **java** tag would be useful.

dakshdeepHERE commented:
@SSwiniarski I've seen the main algorithm.md and there the implementation of algorithm was done in python that's I did it in java.
![image](https://github.com/Codecademy/docs/assets/94860622/6f48331e-ae00-4396-9995-3d10d7331716)

SSwiniarski commented:
I'll defer to the other maintainers: @yangc95, @KTom101, @caupolicandiaz: do you think it's all right for this general example to be written in Java?

dakshdeepHERE commented:
Sure @SSwiniarski because I was preparing the algorithms in Java for future entries. If we device anything else then I will follow that.

KTom101 commented:
@SSwiniarski @dakshdeepHERE I feel it's alright as long as the language is specified. A couple of our [Search Algorithms](https://www.codecademy.com/resources/docs/ai/search-algorithms) (for example [BFS](https://www.codecademy.com/resources/docs/ai/search-algorithms/breadth-first-search)) do the same thing.

dakshdeepHERE commented:
@KTom101 yes I was using java tags in the code. Should I also specify at beginning about the language we are going to use in the algorithm entries or the tag for each code is enough?

KTom101 commented:
@dakshdeepHERE You can write it before the example similar to how we do in other entries. 
You can write "The following shows how to implement ... in Java" filling in the "..." with what you're showing.

dakshdeepHERE commented:
> You can write "The following shows how to implement ... in Java"

Understood I will do this before examples or mention it in the explanations down below it.

dakshdeepHERE commented:
@SSwiniarski so should we continue with java?

SSwiniarski commented:
@dakshdeepHERE Let's go ahead with Java, with the additions @KTom101 suggested.

dakshdeepHERE commented:
Alright @SSwiniarski should I change the tags to java now?

dakshdeepHERE commented:
@SSwiniarski Done. Is there any other changes you want me to do?

SSwiniarski commented:
@dakshdeepHERE Add @KTom101's suggestion by adding sentences like "The following code illustrates this method in Java." before your code blocks.

dakshdeepHERE commented:
@SSwiniarski alright

dakshdeepHERE commented:
@SSwiniarski Can you check the changes now.

SSwiniarski reviewed (approved):
@dakshdeepHERE This looks good for a second review.

caupolicandiaz commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+- In the first recursive cycle, the value of `a % b` will be `150`. Since 150 is smaller than 500, it cannot be divided evenly by 500. Therefore, the remainder is equal to the original number, which is 150. Consequently, the next arguments for `EuclideanOptimized` are `(500, 150)`.
+- The function is restarted with the arguments `(500, 150)`. Upon entering the function, the `if` statement is encountered. However, since `b` is not equal to 0, the program exits the `if` statement. The next recursive cycle will use the arguments `(150, 50)`, which are obtained by calculating the modulus of `500 % 150`.
+- Once again, the function is initiated with arguments `(150, 50)`. Upon entering the function, the `if` statement is encountered. Similarly to previous iterations, `b` is not equal to 0, and thus the program exits the `if` statement. The new arguments for the next recursive cycle are determined by calculating the modulus of `150 % 50`, resulting in `(50, 0)`.
+- In this recursive cycle, when the code enters the `if` statement, the condition `b == 0` is satisfied. Therefore, the function will return `a`, and the highest common factor (HCF) for the input will be 50.
```
```suggestion
- In this recursive cycle, when the code enters the `if` statement, the condition `b == 0` is satisfied. Therefore, the function will return `a`,  which has a value of `50`, the highest common factor for the original arguments.
```

caupolicandiaz commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+- Once again, the function is initiated with arguments `(150, 50)`. Upon entering the function, the `if` statement is encountered. Similarly to previous iterations, `b` is not equal to 0, and thus the program exits the `if` statement. The new arguments for the next recursive cycle are determined by calculating the modulus of `150 % 50`, resulting in `(50, 0)`.
+- In this recursive cycle, when the code enters the `if` statement, the condition `b == 0` is satisfied. Therefore, the function will return `a`, and the highest common factor (HCF) for the input will be 50.
+
+## Time complexities
```
```suggestion
## Time Complexities
```

caupolicandiaz commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+- Method 1:
+
+  - The first code uses a simple iterative approach to find the greatest common divisor (GCD) of two numbers. It starts by finding the minimum value between `a` and `b`, then iterates from that value down to 1, checking if it divides both `a` and `b`. Therefore, the time complexity of this code is **O(min(a, b))**.
```
```suggestion
  - The first example uses a simple iterative approach to find the greatest common divisor (GCD) of two numbers. It starts by finding the minimum value between `a` and `b`, then iterates from that value down to 1, checking if it divides both `a` and `b`. Therefore, the time complexity of this code is *O(min(a, b))*.
```

caupolicandiaz commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+- Method 2:
+
+  - The second code also calculates the GCD using an iterative approach known as the Euclidean algorithm. It repeatedly subtracts the smaller number from the larger number until the two numbers become equal (the GCD). The time complexity of this algorithm depends on the number of iterations required to reach the GCD. In the worst case, where one number is a multiple of the other, the time complex ...
```
```suggestion
  - The second example also calculates the GCD using an iterative approach known as the Euclidean algorithm. It repeatedly subtracts the smaller number from the larger number until the two numbers become equal (the GCD). The time complexity of this algorithm depends on the number of iterations required to reach the GCD. In the worst case, where one number is a multiple of the other, the time complexity is *O(max(a, b))*.
```

caupolicandiaz commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+- Method 3:
+
+  - The third code is an optimized version of the Euclidean algorithm that uses recursion. It calculates the GCD by repeatedly taking the modulus of `a` with `b` and calling itself with the new values (`b` and `a%b`) until `b` becomes 0. The time complexity of this optimized Euclidean algorithm is **O(log(max(a, b)))** since the algorithm reduces the values quickly by taking the modulus.
```
```suggestion
  - The third example is an optimized version of the Euclidean algorithm that uses recursion. It calculates the GCD by repeatedly taking the modulus of `a` with `b` and calling itself with the new values (`b` and `a%b`) until `b` equals `0`. The time complexity of this optimized Euclidean algorithm is *O(log(max(a, b)))* since the algorithm reduces the values quickly by taking the modulus.
```

caupolicandiaz commented on content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md:
```diff
@@ -0,0 +1,149 @@
+
+To summarize, the time complexities of the three codes are as follows:
+
+- Code 1: **O(min(a, b))**
+- Code 2: **O(max(a, b))**
+- Code 3: **O(log(max(a, b)))**
```
```suggestion
- Method 1: *O(min(a, b))*
- Method 2: *O(max(a, b))*
- Method 3: *O(log(max(a, b)))*
```

caupolicandiaz reviewed (changes requested):
@dakshdeepHERE I've completed my initial review. Please update the entry, and tag me when you've made the changes. Thanks

dakshdeepHERE commented:
@caupolicandiaz All the changes are done. Can you check once if there's any other changes needs to be done?

caupolicandiaz reviewed (approved):
@dakshdeepHERE @SSwiniarski I pushed a couple of additional small edits. I think this is good to go.

SSwiniarski commented:
@dakshdeepHERE Congrats on the new entry. Live link here: https://www.codecademy.com/resources/docs/general/algorithm/euclidean-algorithm

![image](https://github.com/Codecademy/docs/assets/86081858/8da4c584-6949-4de0-942d-aa1107d9c682)
